app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -0,0 +1,84 @@
+require 'active_support'
+require 'active_support/core_ext'
+require 'net/ldap'
+require 'json'
+require 'aws-sdk-iam'
+require 'aws-sdk-core'
+
+
+module Authentication
+  module AuthnIam
+
+    class Authenticator
+      
+      
+      def initialize(env:)
+        @env = env
+      end
+
+      def valid?(input)
+        @authn_name, @account, @login, password, @service_id = input.authenticator_name, input.account, input.username, input.password, input.service_id
+
+        # JSON holding the AWS signed headers 
+        @signed_aws_headers = JSON.parse input.password
```

Comment:
Instead of using class variables for passing variables into a method, how about we pass it into the `is_trusted_by_aws?` method?  Doing this will make it easier to test the `is_trusted_by_aws?` in isolation. Maybe something like:
```ruby
signed_aws_headers = JSON.parse(input.password)
is_trusted_by_aws?(signed_aws_headers) && ...
```

Reply:
refactored and cleaned up